### PR TITLE
Mount propagation fixes (and also oomd)

### PR DESCRIPTION
**Pull request by K900**
ugh.

### CODE REVIEW

**Comment by K900** [comment]
The /dev/shm dance is lifted from https://github.com/arkane-systems/bottle-imp/blob/master/binsrc/imp-executor/__main__.py#L28

File: modules/wsl-distro.nix [hunk 1]
```diff
@@ -177,5 +177,10 @@ with lib; {
             '';
           };
         })
+        # this option doesn't exist on older NixOS, so hack.
+        (lib.optionalAttrs (builtins.hasAttr "oomd" options.systemd) {
```
Comment (SuperSandro2000) [inline modules/wsl-distro.nix:181]: Maybe we should just lift the minimal supported version to 22.11

File: modules/wsl-distro.nix [hunk 2]
```diff
@@ -177,5 +177,10 @@ with lib; {
...
+          # OOMd requires cgroup pressure info which WSL doesn't have
```
Comment (SuperSandro2000) [inline modules/wsl-distro.nix:182]:
```suggestion
          # OOMD requires cgroup pressure info which WSL doesn't have
```

File: modules/wsl-distro.nix (hunk 1 above)
Comment (K900) [reply modules/wsl-distro.nix:181]: That is probably better done in a separate PR.

Comment (nzbr) [reply modules/wsl-distro.nix:181]: That'd be #177

Comment (K900) [reply modules/wsl-distro.nix:181]: That's not really dropping support for 22.05 though.

**Pull request merged**